_drafts/conviva-dag-process-time-prod-issue-draft.md

```diff
@@ -0,0 +1,69 @@
+---
+layout: post
+title: "The Problem With Concurrent Hashmaps"
+category: concurrency
+---
+
+## Setting The Stage
+
+We recently faced an interesting production issue at [Conviva](https://www.conviva.com/), related to the time taken by our compute engine to process customer events.
+
+On Feb 2nd, we noticed a recurring issue related to a specific customer where their P99 for processing time would absolutely shoot up. It was strange because we could only see it for a single customer which pointed to the issue most likely having to do with the customer's specific DAG.
```

Comment:
DAG is our own term, but an outside reader wouldn't know what that means.  You could say "processing logic", but we probably need context.  Would be useful to have an extra paragraph talking about our realtime streaming platform, that uses Time State Analytics (link to the paper), and point out this is very high scale, very advanced stateful processing in Rust, and that each customer has slightly different processing logic and rules which is captured by a data flow DAG.  Remember, part of the aim is to sell our platform or that our technology is "cool".

Reply:
@velvia yeah, definitely makes sense. I've asked Gopi & Pawas for numbers like number of events we handle per second or per minute so people can get an idea of the scale.

Let me know if you have any numbers in mind for this.